src/Driver/PDO/Connection.php

```diff
@@ -137,4 +150,29 @@ public function getWrappedConnection(): PDO
 
         return $this->getNativeConnection();
     }
+
+    /**
+     * @return PDOException&DriverException
+     *
+     * @todo this method will be removed in Doctrine DBAL 4.0.
+     */
+    private function makeDriverException(PDOException $exception): DriverException
+    {
+        /** @psalm-suppress MissingImmutableAnnotation */
+        return new class ($exception) extends PDOException implements DriverException {
```

Comment:
Please create a real class for this exception and annotate it as `/** @internal */`.